blockmanager.go

```diff
@@ -1168,6 +1201,39 @@ out:
 	bmgrLog.Trace("Block handler done")
 }
 
+// TxProcessed sends a message that a transaction has been processed by the
+// block manager.
+func (b *blockManager) TxProcessed() {
+	b.txProcessed <- struct{}{}
+}
+
+// WaitForTx waits until a transaction has been processed by the block manager.
+func (b *blockManager) WaitForTx() {
+	<-b.txProcessed
```

Comment:
As mentioned above, this is incorrect.  You're signalling when *any* peer receives a transaction which is not the intention.  The intention is to wait until a specific peer fulfills the request.  For that, the channel needs to be in the peer data.